src/observables/CreateOptionStream.ts

```diff
@@ -0,0 +1,24 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Options } from "../omnisharp/options";
+import { vscode } from "../vscodeAdapter";
+import 'rxjs/add/operator/take';
+import 'rxjs/add/operator/publishBehavior';
+import { Observable } from "rxjs/Observable";
+import { Observer } from "rxjs/Observer";
+
+export default function createOptionStream(vscode: vscode): Observable<Options> {
+    return Observable.create((observer: Observer<Options>) => {
+        let disposable = vscode.workspace.onDidChangeConfiguration(e => {
+            //if the omnisharp or csharp configuration are affected only then read the options
+            if (e.affectsConfiguration('omnisharp') || e.affectsConfiguration('csharp')) {
+                observer.next(Options.Read(vscode));
+            }
+        });
+
+        return () => disposable.dispose();
```

Comment:
This and the line below are beyond my rx-fu, can you explain them?

Reply:
publishBehavior is used to create a behavior observable which has the initial value as Option.Read(). refCount() is used so that the dispose function on the observable is called only when all the observers who susbcribed to this observable invoke unsubscribe. This test verifies this behavior https://github.com/OmniSharp/omnisharp-vscode/pull/2316/files#diff-dd9d0618b9edb503c00be5f531debc27R78